plutus-tx-plugin/src/PlutusTx/Compiler/Kind.hs

```diff
@@ -16,6 +16,24 @@ import qualified PlutusCore              as PLC
 compileKind :: Compiling uni fun m => GHC.Kind -> m (PLC.Kind ())
 compileKind k = withContextM 2 (sdToTxt $ "Compiling kind:" GHC.<+> GHC.ppr k) $ case k of
     -- this is a bit weird because GHC uses 'Type' to represent kinds, so '* -> *' is a 'TyFun'
-    (GHC.isLiftedTypeKind -> True)        -> pure $ PLC.Type ()
-    (GHC.splitFunTy_maybe -> Just (i, o)) -> PLC.KindArrow () <$> compileKind i <*> compileKind o
-    _                                     -> throwSd UnsupportedError $ "Kind:" GHC.<+> GHC.ppr k
+    (GHC.isLiftedTypeKind -> True)         -> pure $ PLC.Type ()
+    (GHC.splitFunTy_maybe -> Just (i, o))  -> PLC.KindArrow () <$> compileKind i <*> compileKind o
+    -- TODO: should construct a proper kind of unboxed tuple
```

Comment:
I think this is right?

Reply:
But what if `(# (# ..., ... #), (# ..., ... #))`? It should construct the same kind for the arguments, isn't?

Reply:
right, but that should also work if we give it kind `Type -> Type -> Type`. I guess this needs to handle unboxed tuples of other arities, though.